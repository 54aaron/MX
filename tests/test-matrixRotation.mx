int main(){
    Matrix n;
    n = [[1,2],[3,4]];
    prints("==============================================");
    prints("Matrix n:");
    print_matrix(n);
    prints("==============================================");
    prints("Matrix 90 Degrees Clockwise");
    print_matrix(transformation(n,5));
    prints("==============================================");
    prints("Matrix 180 Degrees");
    print_matrix(transformation(n,6));
    prints("==============================================");
    prints("Matrix 90 Degrees Anticlockwise:");
    print_matrix(transformation(n,7));
    prints("==============================================");
}
