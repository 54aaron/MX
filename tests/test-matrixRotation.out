==============================================
Matrix n:
[ 1, 2 ]
[ 3, 4 ]

==============================================
Matrix 90 Degrees Clockwise
[ 3, 4 ]
[ -1, -2 ]

==============================================
Matrix 180 Degrees
[ -1, -2 ]
[ -3, -4 ]

==============================================
Matrix 90 Degrees Anticlockwise:
[ -3, -4 ]
[ 1, 2 ]

==============================================
